myTest.java

```diff
@@ -0,0 +1,177 @@
+        ilyaCartFile.deleteOnExit();
+    }
+
+    @AfterEach
+    public void setOriginalSystemOut(){
+        System.setOut(originalOut);
+    }
+
+    @BeforeEach
+    public void setUP()  {
+        ilyaCart = new Cart("ilya-cart");
+
+        car = new RealItem();
+        car.setName("KIA");
+        car.setPrice(20000.0);
+        car.setWeight(1600.0);
+        realItemCar  = "Class: class shop.RealItem; Name: KIA; Price: 20000.0; Weight: 1600.0";
+
+        game = new VirtualItem();
+        game.setName("Star Wars");
+        game.setPrice(100);
+        game.setSizeOnDisk(36000.0);
+        virtualItemGame = "Class: class shop.VirtualItem; Name: Star Wars; Price: 100.0; Size on disk: 36000.0";
+
+        itemsAll = realItemCar + "\r\n" + virtualItemGame + "\r\n";
+
+        ilyaCart.addRealItem(car);
+        ilyaCart.addVirtualItem(game);
+
+        expectedPrice = game.getPrice() + game.getPrice()*TAX + car.getPrice() + car.getPrice()*TAX;
+
+        ilyaCartObjectContent = gson.toJson(ilyaCart);
+
+        System.setOut(new PrintStream(outShowItemsContent));
+    }
+
+    @Test
+    @Order(3)
+    @Tag("Cart")
+    public void testCartName() {
+        assertEquals("ilya-cart", ilyaCart.getCartName());
+    }
+
+    @Test
+    @Order(4)
+    @Tag("Cart")
+    public void testTotalPrice() {
+        assertEquals(expectedPrice, ilyaCart.getTotalPrice());
+    }
+
+    @Test
+    @Order(5)
+    @Tag("Cart")
+    public void testShowItems() {
+        ilyaCart.showItems();
+        assertEquals(itemsAll, outShowItemsContent.toString());
+    }
+
+    @Test
+    @Order(6)
+    @Tag("Cart")
+    public void testDeleteRealItem() {
+        ilyaCart.deleteRealItem(car);
+        ilyaCart.showItems();
+        assertEquals(virtualItemGame+"\r\n", outShowItemsContent.toString());
+    }
+
+    @Test
+    @Order(7)
+    @Tag("Cart")
+    public void testDeleteVirtualItem() {
+        ilyaCart.deleteVirtualItem(game);
+        ilyaCart.showItems();
+        assertEquals(realItemCar+"\r\n", outShowItemsContent.toString());
+    }
+
+    @Test
+    @Tag("RealItem")
+    @Order(1)
+    public void testRealItem(){
+        assertAll("Should return name, weight and price of the car",
+                ()-> assertEquals("KIA",car.getName()),
+                ()-> assertEquals(20000.0, car.getPrice()),
+                ()-> assertEquals(1600.0, car.getWeight())                    ,
+                ()-> assertEquals(realItemCar , car.toString())
+        );
+    }
+
+    @Test
+    @Tag("VirtualItem")
+    @Order(2)
+    public void testVirtualItem(){
+        assertAll("Should return name, weight and price of the game",
+                ()-> assertEquals("Star Wars", game.getName()),
+                ()-> assertEquals(100.0, game.getPrice()),
+                ()-> assertEquals(36000.0, game.getSizeOnDisk()),
+                ()-> assertEquals(virtualItemGame, game.toString())
+        );
+    }

```

Comment:
It was a trap. VirtualItem and RealItem classes were designed as simple models without any business logic. There is no needs to write unit tests for them, especially for getters/settters. The only possible test is for toString method, because it was overriden

Reply:
all assertions for getters/setters were removed.